addons/sale/wizard/sale_order_cancel.py

```diff
@@ -1,20 +1,87 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import UserError
 
 
 class SaleOrderCancel(models.TransientModel):
     _name = 'sale.order.cancel'
-    _description = "Sales Order Cancel"
+    _inherit = 'mail.composer.mixin'
+    _description = 'Sales Order Cancel'
 
+    @api.model
+    def _default_email_from(self):
+        if self.env.user.email:
+            return tools.formataddr((self.env.user.name, self.env.user.email))
+        raise UserError(_("Unable to post message, please configure the sender's email address."))
+
+    @api.model
+    def _default_author_id(self):
+        return self.env.user.partner_id
+
+    # origin
+    email_from = fields.Char(string='From', default=_default_email_from)
+    author_id = fields.Many2one(
+        'res.partner', string='Author', index=True, ondelete='set null', default=_default_author_id
```

Comment:
Sales-payment guideline: string shown to the user should be specified in double quotes :eyes: